src/platform/forms-system/src/js/utilities/data/reduceErrors.js

```diff
@@ -0,0 +1,156 @@
+const ignoreKeys = [
+  'title',
+  'path',
+  'depends',
+  'schema',
+  'chapterTitle',
+  'chapterKey',
+  'pageKey',
+  'type',
+  'required',
+  'initialData',
+];
+
+export const isIgnoredSchemaKey = key =>
+  key.startsWith('ui:') || ignoreKeys.includes(key);
+
+// Find the chapter and page name that contains the property (name) which is
+// used to build a link that will expand the associated accordion when clicked
+const getPropertyInfo = (pageList = [], name, instance = '') => {
+  const findPageIndex = (obj, insideInstance = instance === '') => {
+    if (obj && typeof obj === 'object') {
+      return Object.keys(obj).findIndex(key => {
+        if (isIgnoredSchemaKey(key)) {
+          return false;
+        }
+        if (insideInstance && name === key && obj[name]) {
+          return true;
+        }
+        return findPageIndex(obj[key], insideInstance || key === instance) > -1;
+      });
+    }
+    return -1;
+  };
+  return pageList.find(page => findPageIndex(page) > -1) || {};
+};
+
+/**
+ * @typedef Form~errors
+ * @type {object[]}
+ * @property {string} name - form element name from uiSchema
+ * @property {number|null} index - if the value is an array, this is the index
+ * @property {string} message - processed error message
+ * @property {string} chapterKey - the chapter the element is associated with
+ * @property {string} pageKey - the page within the chapter the element is
+ *   associated with
+ */
+/**
+ * @typedef Form~rawErrors - list of raw errors that are output from jsonschema
+ *   validator
+ * @type {object}
+ * @property {string} property
+ * @property {string} message
+ * @property {string} name
+ * @property {string|number} argument - element name or index in array
+ * @property {object} schema - ignored
+ * @property {string} stack - error message used in processing
+ * @property {string[]} __errors - may contain multiple error messages for one
+ *   element
+ *
+ * There are four types of hardcoded validation error messages:
+ *
+ * min/max: {
+    property: 'instance.someProperty',
+    message: 'does not meet minimum length of 1',
+    name: 'minItems',
+    argument: 1,
+    schema: {...},
+    stack: 'instance.someProperty does not meet minimum length of 1'
+  }
+  * required: {
+    property: 'instance',
+    message: 'requires property "someProperty"',
+    name: 'required',
+    argument: 'someProperty',
+    schema: {...},
+    stack: 'instance requires property "someProperty"'
+  }
+  * array: {
+    property: 'instance.newDisabilities[0]',
+    message: 'requires property "cause"',
+    name: 'required',
+    argument: 'cause',
+    schema: {...},
+    stack: 'instance.newDisabilities[0] requires property "cause"'
+  }
+  * non-empty "__errors" array: {
+    __errors: ['error message']
+  }
+*/
+/**
+ * @typedef Form~formErrors
+ * @type {object}
+ * @property {Form~rawErrors}
+ * @property {Form~errors}
+ */
+/**
+ * Process rawErrors from jsonschema validator into a more friendly list
```

Comment:
"more friendly list" means the resulting list of errors is easy to read. If you look at the `rawErrors`, it's a list of every form page with lots of empty errors (`__errors: []`; here's [an example Sentry log](http://sentry.vfs.va.gov/vets-gov/website-production/issues/12188/events/686c28acac304f72811fb3b05bbcc172/)).

I left the raw errors in the sentry logs and included the processed error list just in case there are errors that were missed in the `reduceErrors` function.

Reply:
Looks great, thanks!